test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -760,14 +684,149 @@ private static void CounterUpdateThread<T>(object obj)
             }
         }
 
+        private static void HistogramUpdateThread<T>(object obj)
+            where T : struct, IComparable
+        {
+            if (obj is not UpdateThreadArguments<T> arguments)
```

Comment:
Why not update the method signature to accept `UpdateThreadArguments<T>` instead of `object`?

Reply:
It looks like `new Thread(HistogramUpdateThread<T>)` will not work if the parameter is not of type `object`. That would be a good change though if anyone knows how to make it. Couldn't find any examples of such in this repository.

Changed it to an `int` here to demonstrate
![image](https://user-images.githubusercontent.com/15165028/145308948-db3d13ed-3bdb-4f0c-ac2b-37b52b4b67db.png)

Reply:
You could try something like this:

```
t[i] = new Thread(() => HistogramUpdateThread<T>(argsToThread));
t[i].Start();
```